server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesResponse.java

```diff
@@ -76,4 +105,158 @@ public void writeTo(StreamOutput out) throws IOException {
             }
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        GetAliasesResponse that = (GetAliasesResponse) o;
+        return Objects.equals(fromListAliasesToSet(aliases), fromListAliasesToSet(that.aliases));
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(fromListAliasesToSet(aliases));
+    }
+
+    private ImmutableOpenMap<String, Set<AliasMetaData>> fromListAliasesToSet(ImmutableOpenMap<String, List<AliasMetaData>> list) {
+        ImmutableOpenMap.Builder<String, Set<AliasMetaData>> builder = ImmutableOpenMap.builder();
+        list.forEach(e -> builder.put(e.key, new HashSet<>(e.value)));
+        return builder.build();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+
+        final boolean namesProvided = (params.param("name") != null);
+        String[] aliasesNames = Strings.EMPTY_ARRAY;
+        final ImmutableOpenMap<String, List<AliasMetaData>> aliasMap = this.aliases;
+
+        if (params.param("name") != null) {
+            String[] result = Strings.splitStringByCommaToArray(params.param("name"));
+            if (false == Strings.isAllOrWildcard(result)) {
+                aliasesNames = result;
+            }
```

Comment:
with my suggestiona above, some of this should go back to the rest action where you can go back to calling e.g. `paramAsStringArrayOrEmptyIfAll` which are not available here.

Reply:
If this goes back to the RestAction then the `name` param will need to be modified ( by getting `_all` and `*` out ) and then simply access it here ? Or did you mean something else ?

Reply:
ignore me :)